geometric_kernels/kernels/geometric_kernels.py

```diff
@@ -99,3 +101,68 @@ def K_diag(self, X: B.Numeric, **parameters) -> B.Numeric:
         weights = self.eigenvalues(**parameters)  # [M, 1]
         Phi = self.eigenfunctions()
         return Phi.weighted_outerproduct_diag(weights, X, **parameters)  # [N,]
+
+
+class MaternIntegratedKernel(BaseGeometricKernel):
```

Comment:
Class description to add; maybe something as here below:

`r"""
    This class computes a Matérn kernel by integrating over the heat kernel [1].

For non-compact manifolds: 
    .. math:: k_{\nu, \kappa, \sigma^2}(x, x') = \int_0^{\infty}  u^{\nu - 1} e^{-\frac{2 \nu}{\kappa^2} u} k_{\infty, \sqrt{2 u}, \sigma^2}(x, x') \d u

For compact manifolds: 
    .. math:: k_{\nu, \kappa, \sigma^2}(x, x') = \int_0^{\infty}  u^{\nu - 1 + d/2} e^{-\frac{2 \nu}{\kappa^2} u} k_{\infty, \sqrt{2 u}, \sigma^2}(x, x') \d u

    References:

    [1] N. Jaquier, V. Borovitskiy, A. Smolensky, A. Terenin, T. Afour, and L. Rozo. Geometry-aware Bayesian Optimization in Robotics using Riemannian Matérn Kernels. CoRL 2021.
    """`